handshake.go

```diff
@@ -156,15 +156,20 @@ func (srv *Server) potentialConnUpgrade(conn net.Conn, reader *buffer.Reader, ve
 		return conn, reader, version, err
 	}
 
-	tlsConfig := tls.Config{
-		Certificates: srv.Certificates,
-		ClientAuth:   srv.ClientAuth,
-		ClientCAs:    srv.ClientCAs,
+	var tlsConfig *tls.Config
+	if srv.TLSConfig != nil {
+		tlsConfig = srv.TLSConfig
+	} else {
+		tlsConfig = &tls.Config{
+			Certificates: srv.Certificates,
+			ClientAuth:   srv.ClientAuth,
+			ClientCAs:    srv.ClientCAs,
+		}
```

Comment:
I like the idea of simply passing a `tls.Config`. I am ok to introduce a breaking change and simply pass the `tls.Config` type directly.